### ISSUE TITLE
BingX: Lastprice Wrong

### ISSUE THREAD

**Issue by nvrossett** [ccxt/ccxt#24036]
### Operating System

Debian

### Programming Languages

JavaScript

### CCXT Version

4.4.20

### Description

![ws](https://github.com/user-attachments/assets/7cf4d6bb-13bc-456b-942c-65aad276a865)

I'm doing an integration with CCXT and I generated a print to explain the problem:

On the left, I have the @lastPrice channel with the official code generated from the CCXT documentation
https://bingx-api.github.io/docs/#/en-us/swapV2/socket/market.html#Subscribe%20to%20latest%20price%20changes
```const WebSocket = require('ws');
const zlib = require('zlib');

const path = "wss://open-api-swap.bingx.com/swap-market"; 
const CHANNEL = {"id":"24dd0e35-56a4-4f7a-af8a-394c7060909c","reqType": "sub","dataType":"BTC-USDT@lastPrice"}; 

let socket;
let receivedMessage = "";

function init() {
  socket = new WebSocket(path);
  socket.on('open', onOpen);
  socket.on('message', onMessage);
  socket.on('error', onError);
  // socket.on('close', onClose);
}

function onOpen() {
  console.log("WebSocket connected");
  socket.send(JSON.stringify(CHANNEL));
}

function onError(error) {
  console.log("WebSocket error:", error);
}

function onMessage(message) {
  const buf = Buffer.from(message);
  const decodedMsg = zlib.gunzipSync(buf).toString('utf-8');
  console.log(decodedMsg);
  if (decodedMsg === "Ping") {
    socket.send('Pong');
    console.log('Pong');
  }
  receivedMessage = decodedMsg;
}

// function onClose() {
//   console.log("WebSocket closed");
// }

init();

```

In it, I have lastPrice synchronized with the broker's version.

What returns in price appears on the exchange

This ticker channel also:
https://bingx-api.github.io/docs/#/en-us/swapV2/socket/market.html#Subscribe%20to%2024-hour%20price%20changes

When I change the header:
From
`const CHANNEL = {"id":"24dd0e35-56a4-4f7a-af8a-394c7060909c","reqType": "sub","dataType":"BTC-USDT@lastPrice"}; `
To
`const CHANNEL = {"id":"24dd0e35-56a4-4f7a-af8a-394c7060909c","reqType": "sub","dataType":"BTC-USDT@ticker"}; `

It brings the correct price

It turns out that using CCXT

Using the following parameter in the connection command:
`options: { 'defaultType': 'swap' }`

Because I'm working on futures markets
In fact, the exchange URL in the print is this:
[https://swap.bingx.com/pt-br/BTC-USDT](https://swap.bingx.com/pt-br/BTC-USDT)

The broker is returning the values ​​from the terminal on the right:
Both in watchTickers and in watchOHLCV, passing the symbol BTC/USDT:USDT

This lack of synchronization simply breaks my application, because it executes actions that it shouldn't because they are not at the price that is seen on the broker.

Notice the timestamp:

The terminal on the left, first it returned the orange square (WS Return: 1)

Then in the terminal on the right it returned the (WS Return: 2)

Then in the terminal on the left it returned the red square again (WS Return: 3)

And the lastPrice is as follows:
```
WS Return 1: 68215.3
WS Return 2: 68196.1 - (WS OHLCV: 68196.1)
WS Return 3: 68215.3
```

I would like to know, is there anything I can do on my side to get the same tick as the broker? Any additional configuration? Another channel that I need to connect to, I don't know.

Because the values ​​are not synchronized with those of the exchange, I initially thought it was just @lastPrice, but while I was writing this I noticed that the @ticker in their example is also correct but not in the Exchange.

I don't know why this happens, I don't know if it's a matter of adjusting the CCXT application, or if it's a configuration problem.

### Code

```
  

```

**Comment by carlosmiei** [comment]
Hello @nvrossett,

CCXT by default uses this channel: https://bingx-api.github.io/docs/#/en-us/swapV2/socket/market.html#Subscribe%20K-Line%20Data


But since the response does not provide an explicit "last price" value, CCXT is using `close` for that field.

**Comment by nvrossett** [comment]
But in this case, in WatchTickers, Last and Close have different values ​​than the exchange. And if I remember correctly, when I was checking yesterday to make the post, the kline that is in the official documentation also has a different value than the exchange. How can I get the same value that is running in the exchange tick?

**Comment by nvrossett** [comment]
@carlosmiei Just correcting, kline returns the same price correctly, I recorded a short video to demonstrate what appears in kline and tickers (left terminal) and what is returning to me in watchTicker and OHLCV, pause the video and compare it with the value that is shown in the exchange, there simply is no value that I can use.

[ws.webm](https://github.com/user-attachments/assets/3ec8d309-561c-4f6f-abf3-4b3481f52173)


To make things easier, here is their example code (responding to the ping to maintain the connection and not generating the coding error for sending an object and then suddenly the ping)

So this makes the websocket return run directly without errors

You already have two channels, one commented to make it easier for you to make the connection

```
const WebSocket = require('ws');
const zlib = require('zlib');

const path = "wss://open-api-swap.bingx.com/swap-market";
// Kline Channel
const CHANNEL = { "id": "e745cd6d-d0f6-4a70-8d5a-043e4c741b40", "reqType": "sub", "dataType": "BTC-USDT@kline_1m" };

// Ticker Channel
// const CHANNEL = { "id": "24dd0e35-56a4-4f7a-af8a-394c7060909c", "reqType": "sub", "dataType": "BTC-USDT@ticker" };

let socket;

function init() {
    socket = new WebSocket(path);
    socket.on('open', onOpen);
    socket.on('message', onMessage);
    socket.on('error', onError);
}

function onOpen() {
    console.log("WebSocket connected");
    socket.send(JSON.stringify(CHANNEL));
}

function onError(error) {
    console.log("WebSocket error:", error);
}

function onMessage(message) {

    // Se a mensagem não for texto simples, tente tratar como compactada
    try {
        const buf = Buffer.from(message);
        const decodedMsg = zlib.gunzipSync(buf).toString('utf-8');
        if (decodedMsg !== 'Ping') {
            const jsonData = JSON.parse(decodedMsg);
            console.log(jsonData);
        } else {
            socket.send('Pong');
        }
    } catch (error) {
        console.log("Error processing message:", error);
    }
}

init();
```

Look at the video, I don't have any value in WatchTicker or OHLCV (both are displayed in the terminal on the right) that refers to the price that is on the exchange.

And as I emphasized, I am using the following parameter in the connection
`options: { 'defaultType': 'swap' }`

By the way, the value shown in watchTickers and OHLCV I did not find in any asset running on the exchange: stop, future, swap with usdt or crypto

**Comment by nvrossett** [comment]
Sorry, a long time ago I had created a block on Bingx from my system, and only now I noticed that it was not reading information from Bingx, but rather from Binance as an "emergency backup connection" for when Bingx stops responding and it continues generating information from indicators and inputs until it starts generating data again.

Let's say it's just a backup connection in case of instability, since I will have aggregated data from several brokers, and I forgot that Bingx was on the ignore list due to necessary adjustments in the past.

Apparently it is now correct.
Sorry for the confusion
